ecobee/auth.go

```diff
@@ -206,6 +208,17 @@ type Client struct {
 // (https://www.ecobee.com/consumerportal/index.html#/dev)
 func NewClient(clientID, authCache string) *Client {
 	return &Client{oauth2.NewClient(
-		context.Background(),
-		TokenSource(clientID, authCache))}
+		context.Background(), TokenSource(clientID, authCache))}
+}
+
+// GetPin retrieves an ecobee Pin and Code, allowing calling code to present them to the user
+// outside of the ecobee request context.
+func GetPin(clientId string) (PinResponse, error) {
```

Comment:
Would Authorize be a better name for this?

Reply:
I'm not sure. This is technically just the first step of a 2-step process to authorize. The "Get" part of the name may be bad, since it's not really a getter, maybe LookupPin or AuthStepOne or something? /shrug

Reply:
Well, 3-step if you count entering the pin into the website...

Reply:
Went with Authorize for the name anyways. Fixed.

Reply:
I chose Authorize based on what the ecobee API calls it.  I agree it's not perfect.

https://www.ecobee.com/home/developer/api/documentation/v1/auth/auth-intro.shtml